File: src/curriculum/two-pointers/TwoSum2/TwoSum.ts
export function twoSum2(numbers: number[], target: number): number[] {
  const cache = new Map<number, number>();

  for (let i = 0; i < numbers.length; i++) {
    const idx1 = cache.get(target - numbers[i]);
    cache.set(numbers[i], i + 1);

    if (idx1) {
      return [idx1, cache.get(numbers[i])!];
    }
  }

  return [];
}

File: src/curriculum/two-pointers/TwoSum2/TwoSum.cpp
#include <iostream>
#include <map>
#include <vector>

int twoSum(std::vector<int> nums, int target)
{
    std::map<int, int> map;

    for (int i = 0; i < nums.size(); i++)
    {
        map.insert(target - nums[i], i + 1);
        std::cout << map[i] << "\n";
        // std::map<int, int>::iterator idx1 = map.find(target - nums[i]);
    }

    return 0;
}

int main()
{
    std::vector<int> numbers;
    numbers[0] = 1;
    numbers[1] = 2;
    twoSum(numbers, 9);
}